test/framework/vsphere.go

```diff
@@ -247,13 +242,34 @@ func (v *VSphere) WithNewVSphereWorkerNodeGroup(name string, workerNodeGroup *Wo
 }
 
 func (v *VSphere) ClusterConfigFillers() []api.ClusterFiller {
-	return v.clusterFillers
+	_, ok := os.LookupEnv(ClusterIPPoolEnvVar)
+	var clusterIP string
+	var err error
+	if ok {
+		clusterIP, err = PopIPFromEnv(ClusterIPPoolEnvVar)
+		if err != nil {
+			v.t.Fatalf("failed to pop cluster ip from test environment: %v", err)
+		}
+	} else {
+		clusterIP = v.generateUniqueIp()
+	}
+
+	return []api.ClusterFiller{api.WithControlPlaneEndpointIP(clusterIP)}
```

Comment:
`v.clusterFillers` can have more than just `WithControlPlaneEndpointIP`, this will break tests with multiple worker node groups

Reply:
I was gonna ask if I needed to append it! thanks for the info. I will update